tests/infra/clients.py

```diff
@@ -413,6 +414,50 @@ def disconnect(self):
         pass
 
 
+class WSClient:
+    def __init__(
+        self,
+        host,
+        port,
+        cert,
+        key,
+        ca,
+        version,
+        format,
+        connection_timeout,
+        request_timeout,
+    ):
+        self.host = host
+        self.port = port
+        self.cert = cert
+        self.key = key
+        self.ca = ca
+        self.format = "json"
+        self.stream = Stream(version, "json")
+        self.request_timeout = request_timeout
+
+    def request(self, request):
+        ws = create_connection(
+            f"wss://{self.host}:{self.port}",
+            sslopt={"certfile": self.cert, "keyfile": self.key, "ca_certs": self.ca},
+        )
+        # TODO: Support sending data over websocket
+        # ws.send(request.to_json())
+        # res = ws.recv()
+        # ws.close()
+        # self.stream.update(rep.content)
+        return request.id
+
+    def signed_request(self, request):
```

Comment:
That's going to need the outer flatbuffer frame to be setup before it can be done.

Reply:
Or, if we support plain JSON-RPC over ws, we can use the old signature scheme.

Reply:
We won't do that, we will only support FlatBuffers over WS, unless there's a clear use-case for something else.